src/image.c

```diff
@@ -82,12 +87,53 @@ void image_thumbnail(struct image* image, size_t size, bool fill,
         offset_y = 0;
     }
 
+    // TODO: probably should be moved outside of this function
+    if (!cache_dir) {
+        cache_dir = getenv("XDG_CACHE_HOME");
+        if (cache_dir) {
+            cache_dir = strcat(cache_dir, "/swayimg");
+        } else {
+            cache_dir = getenv("HOME");
+            if (!cache_dir) {
+                goto thumb_create;
+            }
+            cache_dir = strcat(cache_dir, "/.swayimg");
+        }
+    }
+    sprintf(thumb_path, "%s%s", cache_dir, image->source);
+
+    if (!pixmap_load(&thumb, thumb_path)) {
+        goto thumb_create;
+    }
+
+    if (!(thumb.width == thumb_width && thumb.height == thumb_height)) {
```

Comment:
What if the file has been changed but the image size remains the same?
In this case, we need to recreate the thumbnail.

Reply:
I have no idea how to check this to be honest. Some libnotify thing?

Reply:
I would compare mtime of the files.

Reply:
Then mtime would have to be stored somehow in the thumbnail or another file with mtimes of cached images.
Is there some better method for this?

Reply:
Why not use the file system for this? That's what it's designed for =)

Reply:
Ah, I get it. Compare the mtime of thumbnail and image and if image's mtime is greater than remake the thumbnail, right?

Reply:
It must be ctime

Reply:
One more thing: if antialising mode or fill mode were changed, then we have to recreate thumbnails too.

Reply:
This makes sense, do you have any idea on how to check this?
I could maybe look at config's ctime and recreate each time it is changed, but I don't really like this idea.

Reply:
Store this data within the thumbnail? In meta/exif/...

Reply:
done